src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs

```diff
@@ -0,0 +1,215 @@
+        /// </summary>
+        [Parameter(ParameterSetName = SetByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "SQL Database Managed instance name.")]
+        [Alias("ManagedInstanceName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(ParameterSetName = SetByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// The new SQL administrator password for the Managed instance.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The new SQL administrator password for the Managed instance.")]
+        [ValidateNotNull]
+        public SecureString AdministratorPassword { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Managed instance License Type
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Determines which License Type of Sql Azure Managed Instance to use")]
+        [PSArgumentCompleter(Constants.LicenseTypeBasePrice, Constants.LicenseTypeLicenseIncluded)]
+        public string LicenseType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Storage Size in GB for Managed instance
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Determines how much Storage size to associate with Managed instance")]
+        public int? StorageSizeInGB { get; set; }
+
+        /// <summary>
+        /// Gets or sets the VCore for Managed instance
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Determines how much VCore to associate with Managed instance")]
+        public int? VCore { get; set; }
+
+        /// <summary>
+        /// The tags to associate with the Managed instance.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The tags to associate with the Managed instance.")]
+        [Alias("Tags")]
+        public Hashtable Tag { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether or not to assign identity for Managed instance
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Generate and assign an Azure Active Directory Identity for this Managed instance for use with key management services like Azure KeyVault.")]
+        public SwitchParameter AssignIdentity { get; set; }
+
+        /// <summary>
+        /// AzureSqlManagedInstanceModel object to remove
+        /// </summary>
+        [Parameter(ParameterSetName = SetByInputObjectParameterSet,
```

Comment:
Move inputobject and resourceid up to the top so that the syntax order makes sense.